include/seqan3/search/algorithm/configuration/max_error_rate.hpp

```diff
@@ -35,144 +35,135 @@
+    constexpr max_error_rate(max_error_rate &&)                  noexcept = default;
+    constexpr max_error_rate & operator=(max_error_rate const &) noexcept = default;
+    constexpr max_error_rate & operator=(max_error_rate &&)      noexcept = default;
+    ~max_error_rate()                                            noexcept = default;
+
+    /*!\brief Constructs the object from a set of error specifiers.
+     * \tparam    errors_t A template parameter pack with the error types.
+     * \param[in] errors   A pack of error specifiers.
+     *
+     * \details
+     *
+     * \todo write me
      */
-    template <typename configuration_t, typename ... error_types_t>
+    constexpr max_error_rate(errors_t && ...errors)
     //!\cond
-        requires is_algorithm_configuration_v<remove_cvref_t<configuration_t>>
+        requires sizeof...(errors_t) > 0
     //!\endcond
-    constexpr auto invoke(configuration_t && cfg, error_types_t && ...error_types) const
+        : base_t{}
     {
-        static_assert(is_valid_search_configuration_v<search_cfg::id::max_error_rate, remove_cvref_t<configuration_t>>,
-                      SEQAN3_INVALID_CONFIG(search_cfg::id::max_error_rate));
-
-        std::tuple in{error_types...};
-        std::tuple<search_cfg::total<double>, search_cfg::substitution<double>,
-                   search_cfg::insertion<double>, search_cfg::deletion<double>> out{.0, .0, .0, .0};
-
-        get_type_out<search_cfg::total<double>>(out, in);
-        get_type_out<search_cfg::substitution<double>>(out, in);
-        get_type_out<search_cfg::insertion<double>>(out, in);
-        get_type_out<search_cfg::deletion<double>>(out, in);
-
-        using error_types_list_t = meta::list<remove_cvref_t<error_types_t>...>;
-        constexpr bool total_set = meta::in<error_types_list_t, search_cfg::total<double>>::value;
-        constexpr bool other_error_types_set = meta::count<error_types_list_t, search_cfg::total<double>>::value !=
-                                               meta::size<error_types_list_t>::value;
-
-        double const total_v       {std::get<0>(out)};
-        double const substitution_v{std::get<1>(out)};
-        double const insertion_v   {std::get<2>(out)};
-        double const deletion_v    {std::get<3>(out)};
-
-        if ((0 > total_v     || total_v > 1)     || (0 > substitution_v || substitution_v > 1) ||
-            (0 > insertion_v || insertion_v > 1) || (0 > deletion_v     || deletion_v > 1))
+        //TODO Make to for_each_value once it is in
+        [[maybe_unused]] auto dummy = ((base_t::value[(std::remove_reference_t<errors_t>::_id())] = errors.get()), ..., 0);
+
+        // check correct values.
+        ranges::for_each(base_t::value, [](auto error_elem)
         {
-            throw std::invalid_argument("Error rates must be between 0 and 1.");
-        }
+            if (0.0 > error_elem  || error_elem > 1.0)
```

Comment:
Should we add an epsilon value to avoid failures due to inaccuracies?

Reply:
I think this is fine as is. It is basically just moved to another location.